React/class-components.md

```diff
@@ -0,0 +1,110 @@
+# Class Components
+
+## Learning Goals
+
+- Examine the syntax of creating a class component.
+- Compare that syntax of a functional component with a class component.
+
+## Functional Components
+
+There are two main types of components in React.  _Class Components_ and _Functional Components_.  Until now all the components you created in React were functions like this:
+
+```javascript
+// components/Student.js
+const Student = (props) => {
+  return (
+    <div>
+      <h3>{props.fullname}</h3>
+      <p>{props.email}</p>
+    </div>
+  );
+};
+```
+
+These are called, _functional components,_ because they are composed of one function.  Functional components take in data from `props` and return JSX.  These components are easy to read, test and debug because given a set of props they always return the same JSX back.  However often we want to keep track of information over time, or the _state_ of our component.  This is why functional components are often called _Stateless Components_ in React.  They take props as input and return JSX, which can be a mixture of nested components and HTML.
+
+If you remember back to Ruby, we built objects which combined data, which could change over time, with functionality by defining classes.  We can build React components which can do the same!
+
+## Class Components 
+
+We have already seen a class component `create-react-app` generates one for us called `App`.  
+
+```javascript
+class App extends Component {
+  render() {
+    return (
+      <div className="App">
+       ...
+      </div>
+    );
+  }
+}
+```
+
+This class extends `Component` a class defined in the React library and inherits a bunch of functionality we will go into further in following lectures.  All classes **must** have a `render` method which returns a block of JSX like the functional components we have created thus far.  
+
+### What about `props`
+
+Because a class component is not a single function, any props must be passed into the constructor.
+
+We can see an example by re-writing our Student component as a class.
+
+```javascript
+// components/Student.js
+import React from 'react';
+
+class Student extends React.Component {
+  constructor(props) {
+    super(props);
+  }
```

Comment:
Unless you've got some specific behavior you intend to implement, I would probably not override the constructor here. It makes the code more complex, and doesn't really add to the discussion.